### DISCUSSION TITLE
Arduino IDE Board Selection

### DISCUSSION THREAD

**Discussion by randel0** [hatlabs/discussions#19] [Q&A]
I've done a lot of development in the Arduino IDE environment, and am delighted to begin work on my SH-ESP32-based Mast Rotation interface box.

I'm pointing the Arduino preferences to the ESP32 board definitions from Espressif, but I don't see a choice for the HAT LABS SH-ESP32.  There are many other choices, (Adafruit ESP32 Feather, Nano32, etc.) but the example temp sensor code does not seem to compile on those that I have selected.

Is there a board definition file that I need to include, or can you suggest an existing board definition to try?

Thanks!

**Comment by mairas** [comment]
Hi Randel,

I recommend using Visual Studio Code and PlatformIO for SH-ESP32 development instead. It may be more complex in the beginning but you'll have a much smoother experience managing dependencies etc.

If you insist on using Arduino IDE, search for instructions for building projects for esp32devkit. Those are equivalent wrt. build parameters and flashing.

Best,

Matti Airas

**Reply by randel0** [reply]
Thanks. I'm going to try the Arduino IDE for the time being.

I was able to successfully communicate with the board, and upload "Blink".

FYI, the board profile that worked for me is "DOIT ESP32 DEVKIT I", but only when I modified the upload speed to 115200.
